### PR TITLE
chore: add initial Java runtime testing

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This is the initial runtime setup (for Java only) I was thinking about that can be replicated across all generators to ensure that the generated models are semantically correct by adding tests in Java, using the generated models.

What this PR includes:
- Workflow that runs the Java runtime tests only when related files changes.
- A self-contained Java project with a simple model generation and tests to see it working.

What this enables down the line:
- When we are happy with the manually created tests, we could choose to include them in the generation phase: https://github.com/asyncapi/modelina/pull/839
- This avoids us from releasing new versions that break the promise of the models (i.e. issues such as https://github.com/asyncapi/modelina/issues/1263)
- Reducing the number of generator snapshot tests to focus on the runtime tests. 

**Related issue(s)**
The first step for https://github.com/asyncapi/modelina/issues/612

### CODE REVIEW

**Comment by jonaslagoni** [comment]
@kennethaasan would love to get your initial feedback and suggestions for this setup, I think this is the way to go instead of all those snapshot tests in TypeScript.

What do you think?

File: test/runtime/runtime-java/pom.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+          <!-- It will execute command "npm build" inside "/src/main/javascript" 
+            directory to clean and create "/dist" directory -->
+          <execution>
+            <id>npm build</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>npm</goal>
+            </goals>
+            <configuration>
+              <arguments>run build</arguments>
```
Comment (jonaslagoni) [inline test/runtime/runtime-java/pom.xml:109]: Actually pretty cool that this plugin could be used to install the AsyncAPI CLI and then generate the models with Modelina for Java projects 😄

**Comment by kennethaasan** [comment]
Awesome stuff. I'll have a look at it next week.

**Comment by dlkj** [comment]
The java stuff looks good to me. Great stuff.

**Comment by jonaslagoni** [comment]
Thanks for the reviews! 🔥

**Pull request merged**